pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -431,6 +436,69 @@ protected CompletableFuture<TransactionPendingAckInternalStats> internalGetPendi
+                                                .thenApply(segmentInternalStats -> {
+                                                    SnapshotInternalStats segmentStats = new SnapshotInternalStats();
+                                                    segmentStats.managedLedgerName = segmentTopic
+                                                            .getManagedLedger().getName();
+                                                    segmentStats.managedLedgerInternalStats = segmentInternalStats;
+                                                    transactionBufferInternalStats.segmentInternalStats = segmentStats;
+                                                    return null;
+                                        }))
+                                .thenCompose(ignore -> {
+                                    this.topicName = TopicName.get(TopicDomain.persistent.toString(), namespaceName,
+                                            SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT_INDEXES);
+                                    return getExistingPersistentTopicAsync(authoritative)
+                                            .thenCompose(indexTopic ->  indexTopic.getManagedLedger()
+                                                    .getManagedLedgerInternalStats(metadata)
+                                                    .thenApply(indexInternalStats -> {
+                                                        SnapshotInternalStats indexStats = new SnapshotInternalStats();
+                                                        indexStats.managedLedgerName = indexTopic
+                                                                .getManagedLedger().getName();
+                                                        indexStats.managedLedgerInternalStats = indexInternalStats;
+                                                        transactionBufferInternalStats
+                                                                .segmentIndexInternalStats = indexStats;
+                                                        return null;
+                                                    }));
+                                }).thenApply(ignore -> transactionBufferInternalStats);
+                    } else if (snapshotType == TransactionBuffer.SnapshotType.Single) {
+                        transactionBufferInternalStats.snapshotType = snapshotType.toString();
+                        this.topicName = TopicName.get(TopicDomain.persistent.toString(), namespaceName,
+                                SystemTopicNames.TRANSACTION_BUFFER_SNAPSHOT);
+                        return getExistingPersistentTopicAsync(authoritative)
+                                .thenCompose(singleTopic ->
+                                        singleTopic.getManagedLedger().getManagedLedgerInternalStats(metadata)
+                                                .thenApply(singleInternalStats -> {
```

Comment:
We can `new TransactionBufferInternalStats()` here, just like this:

```java
getManagedLedgerInternalStats.thenApply(mlstats -> {
  TransactionBufferInternalStats txStats = new TransactionBufferInternalStats();
   ...
  return txStats;
});
```

Reply:
Why do we need to `new TransactionBufferInternalStats()` here?
Maybe you mean we need to save a `thenApply`? If so, it looks good to me.

Reply:
> Maybe you mean we need to save a thenApply? If so, it looks good to me.

Yes